src/client/providers/importSortProvider.ts

```diff
@@ -1,57 +1,57 @@
-"use strict";
-
-import * as vscode from "vscode";
-import * as path from "path";
-import * as fs from "fs";
-import * as child_process from "child_process";
-import * as settings from '../common/configSettings';
-import { getTextEditsFromPatch, getTempFileWithDocumentContents } from "../common/editor";
+'use strict';
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { PythonSettings } from '../common/configSettings';
+import { getTempFileWithDocumentContents, getTextEditsFromPatch } from '../common/editor';
 
+// tslint:disable-next-line:completed-docs
 export class PythonImportSortProvider {
-    public sortImports(extensionDir: string, document: vscode.TextDocument): Promise<vscode.TextEdit[]> {
+    public async sortImports(extensionDir: string, document: vscode.TextDocument): Promise<vscode.TextEdit[]> {
         if (document.lineCount === 1) {
-            return Promise.resolve([]);
+            return [];
+        }
+        // isort does have the ability to read from the process input stream and return the formatted code out of the output stream.
+        // However they don't support returning the diff of the formatted text when reading data from the input stream.
+        // Yes getting text formatted that way avoids having to create a temporary file, however the diffing will have.
```

Comment:
I think you have a trailing period that isn't supposed to be there.

Reply:
Fixed